_source/_posts/2022-09-21-angular-dependency-injection.md

```diff
@@ -0,0 +1,296 @@
+---
+layout: blog_post
+title: "Practical Usages of Dependency Injection in Angular"
+author: alisa-duncan
+by: advocate
+communities: [javascript]
+description: "Angular's Dependency Injection is powerful, but complex. In this post, we'll demystify configuring providers in Angular and look at practical examples of using Angular's DI system."
+tags: [angular, javascript, typescript]
+tweets:
+- "Check out this quick overview of Angular's Dependency Injection system for practical examples of how to configure providers! 👀"
+image: blog/angular-dependency-injection/social.jpg
+type: awareness
+---
+
+Angular has an extensive system to add and configure dependencies to the application called providers. When doing this, you use the built-in Dependency Injection (DI) system. This post will cover Angular's powerful DI system at a high level and demonstrate a few practical usages of configuring your dependencies. Let's get practical!
```

Comment:
```suggestion
Angular has an extensive system that uses *providers" to add and configure dependencies to the application you're building. To create providers, you use the built-in Dependency Injection (DI) system. This post will cover Angular's powerful DI system at a high level and demonstrate a few practical use cases and strategies for configuring your dependencies. Let's get practical!
```
 I tried to clarify this opener, but hopefully I didn't misinterpret or change your meaning.